sycl/include/CL/sycl/accessor.hpp

```diff
@@ -275,9 +282,12 @@ class image_accessor
                 "Expected false as Placeholder value for image accessor.");
 
   static_assert(
-      AccessMode == access::mode::read || AccessMode == access::mode::write ||
-          AccessMode == access::mode::discard_write,
-      "Access modes can be only read/write/discard_write for image accessor.");
+      ((IsImageAcc || IsImageArrayAcc) &&
+       (IsImageAccessWriteOnly || IsImageAccessReadOnly)) ||
+          (IsHostImageAcc && (IsImageAccessAnyWrite || IsImageAccessAnyRead)),
+      "Access modes can be only read/write/discard_write for image/image_array "
+      "target accessor, or they can be only "
+      "read/write/discard_write/read_write for host_image target accessor.");
```

Comment:
The assert is essentially checking:
If the accessor is Image Acc Or Image_array Acc, then Access mode can be WriteOnly or ReadOnly.
Or
If the accessor is HostImage Acc, then Access mode can be AnyWrite or AnyRead.
Where the AnyWrite,AnyRead, WriteOnly and ReadOnly are image_accessor defined.